ibis/backends/bigquery/tests/unit/udf/test_core.py

```diff
@@ -0,0 +1,552 @@
+import builtins
+import sys
+import tempfile
+
+import pytest
+
+from ibis.backends.bigquery.udf.core import PythonToJavaScriptTranslator, SymbolTable
+
+
+def test_symbol_table():
+    symbols = SymbolTable()
+    assert symbols["a"] == "let a"
+    assert symbols["a"] == "a"
+
+
+def compile(f):
+    return PythonToJavaScriptTranslator(f).compile()
+
+
+def test_function_def():
+    def f(a, b):
+        return a + b
+
+    expected = """\
+function f(a, b) {
```

Comment:
Are these tests ones that we'd want to parametrize with `snapshot` as well?

Reply:
Yeah, I think so!

Reply:
Done.